### PR TITLE
Support upload_processing_limit

### PR DESCRIPTION
**Pull request by kamil4**
Implements a limit on the number of photos being simultaneously processed during upload.  I expected that adding this would complicate the `process` function but in fact it simplified it (since it was needlessly complicated to begin with).

This implements what was discussed in  https://github.com/LycheeOrg/Lychee/issues/903 and https://github.com/LycheeOrg/Lychee/issues/913. The default limit is `4`. Setting it to `0` disables the limit.

There will be a trivial corresponding server-side PR that adds a config variable.

### CODE REVIEW

**Review by ildyria — APPROVED** [review]
~~Not tested but looks okay~~.

~~Tested, BREAKING : you cannot upload more than one file at once.~~

Fixed.

**Comment by pchampio** [comment]
Also fixes: https://github.com/LycheeOrg/Lychee/issues/443

File: scripts/main/lychee.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ let lychee = {
 
 	album_subtitle_type: "oldstyle",
 
-	upload_processing_limit: 4,
+	upload_processing_limit: 2,
```
Comment (kamil4) [inline scripts/main/lychee.js:53]: Why did you change it to `2`? And why only in this least important place?

File: scripts/main/lychee.js [hunk 2]
```diff
@@ -252,7 +252,7 @@ lychee.init = function () {
 			lychee.admin = !lychee.api_V2;
 			lychee.nsfw_visible_saved = lychee.nsfw_visible;
 
-			lychee.upload_processing_limit = parseInt(data.config.upload_processing_limit);
+			lychee.upload_processing_limit = parseInt(data.config.upload_processing_limit) || 4;
```
Comment (kamil4) [inline scripts/main/lychee.js:255]: This is wrong. This prevents the value of `0` from being used. I don't think we can use this `||` syntax for numerals that can be `0`.

Comment (ildyria) [reply scripts/main/lychee.js:255]: Hummmm indeed, I did not think about this. The problem is that if you send `null` to `parseInt` you get `NaN` which obviously crash all the upload checks afterwards.

File: scripts/main/lychee.js (hunk 1 above)
Comment (ildyria) [reply scripts/main/lychee.js:53]: I was testing. :'D
